x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlPlugin.java

```diff
@@ -79,6 +80,7 @@
 import java.util.function.Supplier;
 
 public class EsqlPlugin extends Plugin implements ActionPlugin {
+    public static final FeatureFlag INLINESTATS_FEATURE_FLAG = new FeatureFlag("esql_inlinestats");
```

Comment:
What's the implication of having both, a snapshot-only capability, and a feature flag? Wouldn't the former suffice?

Reply:
The feature flag is "bigger" than only running in snapshots. You can turn on the flag in a release build and this will work.